src/blocks/packages.rs

```diff
@@ -0,0 +1,343 @@
+
+    let mut package_manager_vec = Vec::new();
+
+    for &package_manager in config.package_manager.iter() {
+        let backend = match package_manager {
+            PackageManager::Apt => Box::new(
+                Apt::new(config.ignore_phased_updates, ignore_updates_regex.clone()).await?,
+            ) as Box<dyn Backend>,
+            PackageManager::Pacman => {
+                Box::new(Pacman::new(ignore_updates_regex.clone()).await?) as Box<dyn Backend>
+            }
+            PackageManager::Aur => Box::new(Aur::new(
+                config.aur_command.clone().unwrap_or_default(),
+                ignore_updates_regex.clone(),
+            )) as Box<dyn Backend>,
+        };
+
+        package_manager_vec.push(backend);
+    }
+
+    loop {
+        let mut package_manager_map: HashMap<&str, u32> = HashMap::new();
+        let mut critical = false;
+        let mut warning = false;
+
+        // Iterate over the all package manager listed in Config
+        for package_manager in &package_manager_vec {
+            let updates = package_manager.get_updates_list().await?;
+            let updates_count = package_manager.get_update_count(&updates).await?;
+
+            package_manager_map.insert(package_manager.name(), updates_count as u32);
+
+            warning |= warning_updates_regex
+                .as_ref()
+                .is_some_and(|regex| package_manager.has_matching_update(&updates, regex));
+            critical |= critical_updates_regex
+                .as_ref()
+                .is_some_and(|regex| package_manager.has_matching_update(&updates, regex));
+        }
+
+        let mut widget = Widget::new();
+
+        let total_count = package_manager_map.values().sum();
+        widget.set_format(match total_count {
+            0 => format_up_to_date.clone(),
+            1 => format_singular.clone(),
+            _ => format.clone(),
+        });
+        widget.set_values(map!(
+            "icon" => Value::icon("update"),
+            "apt" => Value::number(package_manager_map["apt"]),
+            "pacman" => Value::number(package_manager_map["pacman"]),
+            "aur" => Value::number(package_manager_map["aur"]),
+            "total" => Value::number(total_count),
+        ));
+
+        widget.state = match total_count {
+            0 => State::Idle,
+            _ => {
+                if critical {
+                    State::Critical
+                } else if warning {
+                    State::Warning
+                } else {
+                    State::Info
+                }
+            }
+        };
+        api.set_widget(widget)?;
+
+        select! {
+            _ = sleep(config.interval.0) => (),
+            _ = api.wait_for_update_request() => (),
+        }
+    }
+}
+
+#[async_trait]
+pub trait Backend {
+    fn name(&self) -> &str;
+
+    async fn get_updates_list(&self) -> Result<String>;
+
+    async fn get_update_count(&self, updates: &str) -> Result<usize>;
```

Comment:
I realize that this is similar to how pacman and apt used to work, but since we are rewriting this, maybe let's change `get_updates_list` to return a list of strings (with backend-specific filtering applied, such as phased updates in apt). This way we don't need `get_update_count` at all. This will also make `ignore_updates_regex` option easier to implement in backend-agnostic way.

Reply:
I'll try to do

Reply:
I am sorry, but I am not able to do this

Reply:
May I ask why?

Reply:
> May I ask why?

Aah, I am not able to think how can i do that
I did try something but i think that's wrong so I dropped the idea

Reply:
Maybe, I'll just directly commit, push the code

Reply:
Did it